matter_server/server/device_controller.py

```diff
@@ -274,21 +272,28 @@ async def commission_with_code(
                 attempts,
                 MAX_COMMISSION_RETRIES,
             )
-            result: (
-                PyChipError | None
-            ) = await self._chip_device_controller.commission_with_code(
-                node_id,
-                code,
-                DiscoveryType.DISCOVERY_NETWORK_ONLY
-                if network_only
-                else DiscoveryType.DISCOVERY_ALL,
-            )
-            if result and result.is_success:
-                break
-            if attempts >= MAX_COMMISSION_RETRIES:
-                raise NodeCommissionFailed(
-                    f"Commission with code failed for node {node_id}."
+            try:
+                commissioned_node_id: int = (
+                    await self._chip_device_controller.commission_with_code(
+                        node_id,
+                        code,
+                        DiscoveryType.DISCOVERY_NETWORK_ONLY
+                        if network_only
+                        else DiscoveryType.DISCOVERY_ALL,
+                    )
                 )
+                # We use SDK default behavior which always uses the commissioning Node ID in the
+                # generated NOC. So this should be the same really.
+                LOGGER.info(
+                    "Commissioned Node ID: %s vs %s", commissioned_node_id, node_id
+                )
+                assert commissioned_node_id == node_id
```

Comment:
maybe raise a runtime error here or another somewhat more explicit exception instead of an assert ?
I use assert only when its 100% safe to assume the value should exist (often just to satitsfy the type checker)